src/main/java/org/broadinstitute/hellbender/utils/downsampling/PositionalDownsampler.java

```diff
@@ -0,0 +1,143 @@
+
+    private final ReservoirDownsampler reservoir;
+
+    private final SAMFileHeader header;
+
+    private GATKRead previousRead;
+
+    private List<GATKRead> finalizedReads;
+
+    /**
+     * Construct a PositionalDownsampler
+     *
+     * @param targetCoverage Maximum number of reads that may share any given alignment start position. Must be > 0
+     * @param header SAMFileHeader to use to determine contig ordering. Non-null.
+     */
+    public PositionalDownsampler( final int targetCoverage, final SAMFileHeader header ) {
+        Utils.validateArg(targetCoverage > 0, "targetCoverage must be > 0");
+        Utils.nonNull(header);
+
+        this.reservoir = new ReservoirDownsampler(targetCoverage);
+        this.finalizedReads = new ArrayList<>();
+        this.header = header;
+        clearItems();
+        resetStats();
+    }
+
+    @Override
+    public void submit( final GATKRead newRead ) {
+        // If we've moved to a new position, finalize the reads currently in our reservoir.
+        handlePositionalChange(newRead);
+
+        // Pass-through reads that have no assigned position, to avoid downsampling all unmapped reads
+        // to the targetCoverage. Unmapped reads that do have an assigned position *will* be subject to
+        // downsampling, however.
+        if ( ReadUtils.readHasNoAssignedPosition(newRead) ) {
+            finalizedReads.add(newRead);
+        }
+        else {
+            final int reservoirPreviouslyDiscardedItems = reservoir.getNumberOfDiscardedItems();
+            reservoir.submit(newRead);
+            numDiscardedItems += reservoir.getNumberOfDiscardedItems() - reservoirPreviouslyDiscardedItems;
+        }
+
+        previousRead = newRead;
+    }
+
+    private void handlePositionalChange( final GATKRead newRead ) {
+        // Use ReadCoordinateComparator to determine whether we've moved to a new start position.
+        // ReadCoordinateComparator will correctly distinguish between purely unmapped reads and unmapped reads that
+        // are assigned a nominal position.
+        if ( previousRead != null && ReadCoordinateComparator.compareCoordinates(previousRead, newRead, header) != 0 ) {
+            if ( reservoir.hasFinalizedItems() ) {
+                finalizeReservoir();
+            }
+        }
+    }
+
+    private void finalizeReservoir() {
+        finalizedReads.addAll(reservoir.consumeFinalizedItems());
+        reservoir.resetStats();
+    }
+
+    @Override
+    public boolean hasFinalizedItems() {
+        return ! finalizedReads.isEmpty();
+    }
+
+    @Override
+    public List<GATKRead> consumeFinalizedItems() {
+        final List<GATKRead> toReturn = finalizedReads;
+        finalizedReads = new ArrayList<>();
+        return toReturn;
+    }
+
+    @Override
+    public boolean hasPendingItems() {
+        // The finalized items in the ReservoirDownsampler are pending items from the perspective of the
```

Comment:
The `ReservoirDownsampler` considers reads that have been selected for inclusion in the reservoir at any given time to be "finalized", since they have survived the *reservoir* downsampling process. The enclosing `PositionalDownsampler` considers these same reads to be "pending", since they have not yet survived the *positional* downsampling process.